### PR TITLE
added changes to switch hands function

### PR DESCRIPTION
**Pull request by midspooj**
changed the switch_hands code to include the following features:
1. made sure the scoring is calculated correctly per round per player 
2. the code was fixed to output the top scoring player per round and winner of the game
3. adjusted the code to correctly display cards on the table at every turn and round

### CODE REVIEW

**Comment by dani-jimlar** [comment]
lets make sure we can adjust the final score functions

File: src/switch_hands_issues.py [hunk 1]
```diff
@@ -336,50 +298,89 @@ def conduct_round(self):
                 print(
                     f"Player 2's table: {[str(card) for card in self.table.player2_table]}"
                 )
-
-            final_score1 = self.player1.calculate_final_score(
-                self.final_table, first_card_1
-            )
-            final_score2 = self.player2.calculate_final_score(
-                self.final_table, first_card_2
+        def calculate_final_score(self, table):
```
Comment (simrunsharma) [inline src/switch_hands_issues.py:301]: this still not calculating final score correctly.

File: src/switch_hands_issues.py [hunk 2]
```diff
@@ -336,50 +298,89 @@ def conduct_round(self):
...
+            maki_score = 0
+            nigiri_score = 0
+            tempura_pairs = 0
+            sashimi_count = 0
+            dumpling_count = 0
+
+            has_wasabi = False
+            highest_nigiri_score = 0
+
+            for card in table:
```
Comment (simrunsharma) [inline src/switch_hands_issues.py:313]: adjust this so that final_table is shown.

**Review by simrunsharma — APPROVED** [review]
I will change the final table print statment and the recalculate the final score

**Pull request merged**